syntax = "proto3";

package modbus;

// Modbus Bronze Record Schema
// Matches the Databricks Unity Catalog table schema for Modbus data
message ModbusBronzeRecord {
    // Timestamps in microseconds (int64 for large values)
    int64 event_time = 1;
    int64 ingest_time = 2;

    // Source information
    string source_name = 3;

    // Modbus specific fields
    int32 slave_id = 4;
    int32 register_address = 5;
    string register_type = 6;  // holding, input, coil, discrete

    // Sensor identification
    string industry = 7;
    string sensor_name = 8;
    string sensor_path = 9;

    // Value information
    string sensor_type = 10;
    string unit = 11;
    double raw_value = 12;
    int32 scaled_value = 13;
    int32 scale_factor = 14;
}
